src/NServiceBus.RabbitMQ/Administration/QueueCreator.cs

```diff
@@ -21,14 +21,23 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
             using (var connection = connectionFactory.CreateAdministrationConnection())
             using (var channel = connection.CreateModel())
             {
-                foreach (var receivingAddress in queueBindings.ReceivingAddresses)
+                var queueDeclaringTopology = routingTopology as IDeclareQueues;
+                if (queueDeclaringTopology != null)
                 {
-                    CreateQueueIfNecessary(channel, receivingAddress);
+                    queueDeclaringTopology.DeclareAndInitialize(
```

Comment:
I try to keep all lines less than 120 chars long. The reason is that any longer than that, and you have to start horizontal scrolling when looking at diffs in GitHub and I have a pet hate of that. 😄 

That said, I get over-zealous with these kinds of things now and then: I don't mind removing these line breaks if you prefer.